### DISCUSSION TITLE
Discussion for Dega Lite docker-compose file

### DISCUSSION THREAD

**Discussion by vsumit89** [factly/dega#744] [General]
Services in Dega - 

1. Postgres(optional) - We can use sqlite instead of postgres for the light version.
2. Minio(mandatory) - It is used for storing images so for that reason I think it will be mandatory.
3. Iframely(optional) -  Iframely is mainly used in editor components for embedding. We can have an environment variable which can control whether the embed feature is there or not in the editor. 
4. Imgproxy(optional) - We use imgproxy for getting the proxy URL of the image. There are 2 kinds of URL for the image - raw and proxy. We can use the raw url of the image too. But for that we have to change the studio code.
5. Companion(mandatory) - companion is needed for the uppy component in the studio which we use for uploading images.
6. Mailslurper(optional) - this service is a dependency for kratos. It acts as a proxy email client which has all the emails that are sent by kratos. So we can remove it.
7. Meilisearch (optional) - If we remove this then we have to remove all the search bars and make some changes on the server side too. It is optional but removing it might take some time.
8. Keto, Kratos and Oathkeeper(mandatory) - All these services are dependencies for authentication, authorization and session management for dega as well as kavach.  
9. Keto-migrate and kratos-migrate (mandatory) - these services are dependencies for Keto and Kratos, so they are required.
10. Redis (Optional) - we can make caching optional.
11. Kavach (web, server, migrate)(mandatory) - Kavach will manage the roles, policies, tokens and spaces for which it is required.
12. Hukz, Hukz-migrate and Nats (optional) - Depends on whether the end user wants notifications or not. We have an environment which controls the hukz.
13. Dega (templates) (optional)-  It is still in the development stage and we are still experimenting with it so we can make it optional for now.
14. Dega (studio, server, migrate, api) - (Mandatory)

All the services that are marked as mandatory should be in the dega-lite version. We can remove the services which are marked as optional. Out of those I think searching will be helpful for the end user so we should keep meilisearch in the dega-lite version.

**Comment by deshetti** [comment]
@vsumit89 I thought if we disable Meilisearch we use SQL Queries for searching and filtering if Meili is not enabled. Is that not implemented? https://github.com/factly/dega

**Reply by shreeharsha-factly** [reply]
@deshetti We have a search endpoint to get results on all entities. Only that particular endpoint uses only meili and not SQL queries.

**Reply by deshetti** [reply]
@shreeharsha-factly If that is the case, can we disable that search endpoint if Meili is not enabled? As long as the search within entities works, that should be fine. And we will have to mention the same in the documentation.
